pages/dao/[symbol]/proposal/components/instructions/Saber/Deposit.tsx

```diff
@@ -0,0 +1,416 @@
+    .required('Minimum Pool Token Amount is required'),
+})
+
+async function checkInitTokenAccount(
+  account: PublicKey,
+  instructions: TransactionInstruction[],
+  connection: any,
+  mint: PublicKey,
+  owner: PublicKey,
+  feePayer: PublicKey
+) {
+  const accountInfo = await connection.current.getAccountInfo(account)
+  if (accountInfo && accountInfo.lamports > 0) {
+    return
+  }
+  instructions.push(
+    Token.createAssociatedTokenAccountInstruction(
+      ASSOCIATED_TOKEN_PROGRAM_ID, // always ASSOCIATED_TOKEN_PROGRAM_ID
+      TOKEN_PROGRAM_ID, // always TOKEN_PROGRAM_ID
+      mint, // mint
+      account, // ata
+      owner, // owner of token account
+      feePayer
+    )
+  )
+}
+
+const Deposit = ({
+  index,
+  governance,
+}: {
+  index: number
+  governance: ProgramAccount<Governance> | null
+}) => {
+  const connection = useWalletStore((s) => s.connection)
+  const wallet = useWalletStore((s) => s.current)
+  const shouldBeGoverned = !!(index !== 0 && governance)
+  const [formErrors, setFormErrors] = useState({})
+  const { handleSetInstructions } = useContext(NewProposalContext)
+  const { assetAccounts } = useGovernanceAssets()
+
+  const [pool, setPool] = useState<Pool | null>(null)
+
+  const [
+    associatedTokenAccounts,
+    setAssociatedTokenAccounts,
+  ] = useState<null | {
+    A: {
+      account: PublicKey
+      uiBalance: string
+    }
+    B: {
+      account: PublicKey
+      uiBalance: string
+    }
+  }>(null)
+
+  const [form, setForm] = useState<SaberPoolsDepositForm>({
+    assetAccount: undefined,
+    uiTokenAmountA: 0,
+    uiTokenAmountB: 0,
+    uiMinimumPoolTokenAmount: 0,
+  })
+
+  const handleSetForm = ({ propertyName, value }) => {
+    setFormErrors({})
+    setForm({ ...form, [propertyName]: value })
+  }
+
+  useEffect(() => {
+    ;(async () => {
+      if (!form.assetAccount?.pubkey || !form.assetAccount.extensions.token) {
+        return
+      }
+
+      if (!pool) {
+        setAssociatedTokenAccounts(null)
+        return
+      }
+
+      const sourceAccount = form.assetAccount.extensions.token.account.owner
+      const assetTokenMint = form.assetAccount.extensions.token?.account.mint
+
+      // We check if the asset account selected has at least one token common to the selected pool
+      if (
+        !assetTokenMint.equals(pool.tokenAccountA.tokenMint) &&
+        !assetTokenMint.equals(pool.tokenAccountB.tokenMint)
+      ) {
+        setAssociatedTokenAccounts(null)
+        return
+      }
+
+      const [sourceA] = findATAAddrSync(
+        sourceAccount,
+        pool.tokenAccountA.tokenMint
+      )
+
+      const [sourceB] = findATAAddrSync(
+        sourceAccount,
+        pool.tokenAccountB.tokenMint
+      )
+
+      const [resA, resB] = await Promise.allSettled([
+        connection.current.getTokenAccountBalance(sourceA),
+        connection.current.getTokenAccountBalance(sourceB),
+      ])
+      let amountA = ''
```

Comment:
I believe amountA and amountB should be `number | null`. Makes more sense.

Reply:
I agree it makes more sense. I'll change it to that, tho for the record I kinda prefer to work with strings when I have to deal with nasty floats